### ISSUE TITLE
Error in the documentation for cv::getRectSubPix.

### ISSUE THREAD

**Issue by csukuangfj** [opencv/opencv#9788]
##### System information (version)

- OpenCV => :master:

##### Detailed description

The documentation for `cv::getRectSubPix` is outdated.

Please refer to https://github.com/opencv/opencv/blob/master/modules/imgproc/include/opencv2/imgproc.hpp#L2471

There is no `dst` in the function parameters.

The documentation says it supports to specify the border type, but the
function itself currently does not support it, as there is no such a parameter
in the function declaration.

In addition, it does not specify the number of channels supported. But
in the implementation, it requires a single channel or 3-channel input.

Please refer to
https://github.com/opencv/opencv/blob/master/modules/imgproc/src/samplers.cpp#L374

**Comment by fakabbir** [comment]
It seems to be outdated but still usefull, if we look at the code
```c++
CV_IMPL void
cvGetRectSubPix( const void* srcarr, void* dstarr, CvPoint2D32f center )
{
    cv::Mat src = cv::cvarrToMat(srcarr);
    const cv::Mat dst = cv::cvarrToMat(dstarr);
    CV_Assert( src.channels() == dst.channels() );

    cv::getRectSubPix(src, dst.size(), center, dst, dst.type());
}
```
and 
```c++
void cv::getRectSubPix( InputArray _image, Size patchSize, Point2f center,
                       OutputArray _patch, int patchType )
```

`CV_IMPL void cvGetRectSubPix` uses src and dst notation. So will it be good to total remove `dst` ?

Also  indication for sinlge/three channel is needed as the assertion takes place

`CV_Assert( cn == 1 || cn == 3 );`

**Comment by RitikaGupta8734** [comment]
Hi I would like to work upon the issue. Can someone guide me ?

**Comment by mshabunin** [comment]
It seems the issue have been fixed already: #9871
